File: java/programacion_orientada_a_objetos/agenda_con_objetos.java
package programacion_orientada_a_objetos;
import java.util.ArrayList;
class Contacto {
    String nombre;
    String telefono;
    String correo;
    public Contacto(String nombre, String telefono, String correo) {
        this.nombre = nombre;
        this.telefono = telefono;
        this.correo = correo;
    }
    public void mostrarContacto() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Teléfono: " + telefono);
        System.out.println("Correo: " + correo);
        System.out.println("-------------------");
    }
}
public class agenda_con_objetos {
    ArrayList<Contacto> contactos = new ArrayList<>();
    public void agregarContacto(Contacto contacto) {
        contactos.add(contacto);
    }
    public void mostrarAgenda() {
        System.out.println("Agenda de contactos:");
        if (contactos.isEmpty()) {
            System.out.println("La agenda está vacía.");
        } else {
            for (Contacto c : contactos) {
                c.mostrarContacto();
            }
        }
    }
    public static void main(String[] args) {
        agenda_con_objetos agenda = new agenda_con_objetos();
        Contacto c1 = new Contacto("Genis", "3001234567", "genis@email.com");
        Contacto c2 = new Contacto("Maria", "3019876543", "maria@email.com");
        Contacto c3 = new Contacto("Carlos", "3024567890", "carlos@email.com");
        agenda.agregarContacto(c1);
        agenda.agregarContacto(c2);
        agenda.agregarContacto(c3);
        agenda.mostrarAgenda();
    }
}

File: python/agenda_con_objetos.py
class Contacto:
    def __init__(self, nombre, telefono, email):
        self.nombre = nombre
        self.telefono = telefono
        self.email = email
    def mostrar(self):
        return f"Nombre: {self.nombre}, Teléfono: {self.telefono}, Email: {self.email}"
class Agenda:
    def __init__(self):
        self.contactos = []
    def agregar_contacto(self, contacto):
        if isinstance(contacto, Contacto):
            self.contactos.append(contacto)
            print("Contacto agregado correctamente.")
        else:
            print("Error: Debe ser un objeto de tipo Contacto.")
    def mostrar_agenda(self):
        if not self.contactos:
            print("La agenda está vacía.")
        else:
            for contacto in self.contactos:
                print(contacto.mostrar())
    def buscar_contacto(self, nombre):
        for contacto in self.contactos:
            if contacto.nombre.lower() == nombre.lower():
                return contacto
        return None
    def eliminar_contacto(self, nombre):
        contacto = self.buscar_contacto(nombre)
        if contacto:
            self.contactos.remove(contacto)
            print("Contacto eliminado.")
        else:
            print("Contacto no encontrado.")